src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java

```diff
@@ -0,0 +1,267 @@
+
+  /**
+   * Constructs a {@code SelectStatementHandler} with the specified {@link DynamoDbClient} and a new
+   * {@link TableMetadataManager}
+   *
+   * @param client {@code DynamoDbClient}
+   * @param metadataManager {@code TableMetadataManager}
+   */
+  public SelectStatementHandler(DynamoDbClient client, TableMetadataManager metadataManager) {
+    super(client, metadataManager);
+  }
+
+  @Override
+  public List<Map<String, AttributeValue>> handle(Operation operation) throws ExecutionException {
+    checkArgument(operation, Get.class, Scan.class);
+
+    try {
+      if (operation instanceof Get) {
+        GetItemResponse response = executeGet((Get) operation);
+        if (response.hasItem()) {
+          return Arrays.asList(response.item());
+        } else {
+          return Collections.emptyList();
+        }
+      } else {
+        // convert to a mutable list for the Scanner
+        return new ArrayList<>(executeQuery((Scan) operation).items());
+      }
+    } catch (DynamoDbException e) {
+      throw e;
+    }
+  }
+
+  private GetItemResponse executeGet(Get get) {
+    DynamoOperation dynamoOperation = new DynamoOperation(get, metadataManager);
+
+    GetItemRequest.Builder builder =
+        GetItemRequest.builder()
+            .tableName(dynamoOperation.getTableName())
+            .key(dynamoOperation.getKeyMap());
+
+    if (!get.getProjections().isEmpty()) {
+      builder.projectionExpression(String.join(",", get.getProjections()));
+    }
+
+    if (get.getConsistency() != Consistency.EVENTUAL) {
+      builder.consistentRead(true);
+    }
+
+    return client.getItem(builder.build());
+  }
+
+  private QueryResponse executeQuery(Scan scan) {
+    DynamoOperation dynamoOperation = new DynamoOperation(scan, metadataManager);
+    QueryRequest.Builder builder = QueryRequest.builder().tableName(dynamoOperation.getTableName());
+
+    setConditions(builder, scan);
+
+    if (!scan.getOrderings().isEmpty()) {
+      scan.getOrderings()
+          .forEach(
+              o -> {
+                if (dynamoOperation.isSortKey(o.getName())
+                    && o.getOrder() == Scan.Ordering.Order.DESC) {
+                  builder.scanIndexForward(false);
+                }
+              });
+    }
+
+    if (scan.getLimit() > 0) {
+      builder.limit(scan.getLimit());
+    }
+
+    if (!scan.getProjections().isEmpty()) {
+      builder.projectionExpression(String.join(",", scan.getProjections()));
+    }
+
+    if (scan.getConsistency() != Consistency.EVENTUAL) {
+      builder.consistentRead(true);
+    }
+
+    return client.query(builder.build());
+  }
+
+  private void setConditions(QueryRequest.Builder builder, Scan scan) {
+    List<String> conditions = new ArrayList<>();
+
+    conditions.add(getPartitionKeyCondition());
+
+    boolean isRangeEnabled = setRangeCondition(scan, conditions);
+    if (!isRangeEnabled) {
```

Comment:
Sorry, I'm not really sure why `range` is specially treated in here and [below](https://github.com/scalar-labs/scalardb/pull/84/files#diff-fd1d1c3e8897b8263f236fe2b59f5ff7R132).
Could you elaborate on this?

Reply:
At first, I tried to use multiple conditions to specify the range like `:ck > :start_val AND :ck < :end_val`.
However, it didn't work because a condition string can include only 1 condition per key.
(If it could work, we can use `exclusive` option in DynamoDB)
That's why the range condition is specified by `a BETWEEN b AND c `.